src/Migrator/Migrator.php

```diff
@@ -0,0 +1,267 @@
+use Quidco\DbSampler\Cleaner\RowCleaner;
+use Quidco\DbSampler\Collection\TableCollection;
+use Quidco\DbSampler\Collection\ViewCollection;
+use Quidco\DbSampler\ReferenceStore;
+use Quidco\DbSampler\Sampler\Sampler;
+use Quidco\DbSampler\SamplerMap\SamplerMap;
+use Quidco\DbSampler\Writer\Writer;
+
+/**
+ * Migrator class to handle all migrations in a set
+ */
+class Migrator
+{
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var Connection
+     */
+    private $sourceConnection;
+
+    /**
+     * @var Connection
+     */
+    private $destConnection;
+
+    /**
+     * @var ReferenceStore
+     */
+    private $referenceStore;
+
+    private $customCleaners = [];
+
+
+    public function __construct(
+        Connection $sourceConnection,
+        Connection $destConnection,
+        LoggerInterface $logger
+    ) {
+        $this->sourceConnection = $sourceConnection;
+        $this->destConnection = $destConnection;
+        $this->logger = $logger;
+        $this->referenceStore = new ReferenceStore();
+    }
+
+    /**
+     * Perform the configured migrations
+     *
+     * @throws \Exception Rethrows any exceptions after logging
+     */
+    public function execute(string $setName, TableCollection $tableCollection, ViewCollection $viewCollection): void
+    {
+        foreach ($tableCollection->getTables() as $table => $migrationSpec) {
+            // @todo: it'd probably be better to have a proper `migrationspec` config object
+            // rather than relying on properties being present in the json / stdClass object
+
+            $sampler = $this->buildTableSampler($migrationSpec, $table);
+            $writer = new Writer($migrationSpec, $this->destConnection);
+            $cleaner = new RowCleaner($migrationSpec);
+
+            foreach ($this->customCleaners as $alias => $customCleaner) {
+                $cleaner->registerCleaner($customCleaner, $alias);
+            }
+
+            try {
+                $this->ensureEmptyTargetTable($table, $this->sourceConnection, $this->destConnection);
+                $rows = $sampler->execute();
+
+                foreach ($rows as $row) {
+                    $writer->write($table, $cleaner->cleanRow($row));
+                }
+                $writer->postWrite();
+
+                $this->logger->info("$setName: migrated '$table' with '" . $sampler->getName() . "': " . \count($rows) . " rows");
+            } catch (\Exception $e) {
+                $this->logger->error(
+                    "$setName: failed to migrate '$table' with '" . $sampler->getName() . "': " . $e->getMessage()
+                );
+                throw $e;
+            }
+        }
+
+        foreach ($viewCollection->getViews() as $view) {
+            $this->migrateView($view, $setName);
+        }
+
+        $this->migrateTableTriggers($setName, $tableCollection);
+    }
+
+    public function registerCustomCleaner(FieldCleaner $cleaner, string $alias): void
```

Comment:
ideally we'd be able to remove this in a future PR. i have a suspicion that some extra work on refactoring the `Migrator` and `App` classes (and adding a well-designed `Config` object) will show us a better path